PowerEditor/src/MISC/PluginsManager/Notepad_plus_msgs.h

```diff
@@ -1237,10 +1237,28 @@ enum Platform { PF_UNKNOWN, PF_X86, PF_X64, PF_IA64, PF_ARM64 };
 	//scnNotification->nmhdr.hwndFrom = hwndNpp;
 	//scnNotification->nmhdr.idFrom = 0;
 
-	#define NPPN_CMDLINEPLUGINMSG (NPPN_FIRST + 28)  // To notify plugins that the new argument for plugins (via '-pluginMessage="YOUR_PLUGIN_ARGUMENT"' in command line) is available
+	#define NPPN_CMDLINEPLUGINMSG (NPPN_FIRST + 28)  // To notify plugins that there are plugin arguments pluginMessage (wchar_t*) is available
 	//scnNotification->nmhdr.code = NPPN_CMDLINEPLUGINMSG;
 	//scnNotification->nmhdr.hwndFrom = hwndNpp;
 	//scnNotification->nmhdr.idFrom = pluginMessage; //where pluginMessage is pointer of type wchar_t
+	// 
+	// User can pass arguments to plugins via command line argument using:
+	// -pluginMessage="PLUGIN1_ARG1=V1;PLUGIN1_ARG2=V2;PLUGIN2_ARG=V;..."
+	// 
+	// The full string (wchar_t*) will be delivered to all plugins via the NPPN_CMDLINEPLUGINMSG notification. Each plugins can parse and extract the arguments relevant to itself.
+	// 
+	// To avoid the collisions among plugins, the following protocol should be followed:
+	// 1. Each plugin must use its unique namespace (its folder name in plugins directory) as a prefix for its argument names.
+	// 2. The symbol ';' must be used as the delimiter between arguments when there are 2 or more arguments in the pluginMessage string.

```

Comment:
I've been thinking about a corner case: what if someone wanted to pass `;` as an argument?
For example:
```
-pluginMessage="replace_what=, ;replace_with=; ;relpace_what=\t;replace_with=    "
```

Reply:
Plugin author can use any symbol as place-holder of ';' and process it from the plugin side.